release_docs/RELEASE.txt

```diff
@@ -47,6 +47,11 @@ New Features
 
     Configuration:
     -------------
+    - Running H5detect is no longer required for cross-compiling
+
+      The functionality of H5detect is now exercised at library startup,
+      so H5detect has been removed.
```

Comment:
Of course you should mention that this change also eliminates the three CMake variables: HDF5_USE_PREGEN, HDF5_USE_PREGEN_DIR, and HDF5_BATCH_H5DETECT